tuf_conformance/conftest.py

```diff
@@ -0,0 +1,35 @@
+import os
+import pytest
+
+from tuf_conformance.client_runner import ClientRunner
+from tuf_conformance.simulator_server import SimulatorServer
+
+def pytest_addoption(parser) -> None:
+    """Add `--entrypoint` and `--skip-signing` flags to CLI."""
+    parser.addoption(
+        "--entrypoint",
+        action="store",
+        help="the command to invoke the tuf client under test",
+        required=True,
+        type=str,
+    )
+
+@pytest.fixture
+def server():
+    """
+    Parametrize each test with the server under test.
+    """
+    return SimulatorServer(9001)
```

Comment:
the pytest fixture idea sounds good but I think we want a single SimulatorServer: otherwise once we have more than one test we'll need dynamic ports and there will be linear startup costs... So server() should return a singleton.

Reply:
I changed it to use a `yield` which allows us to clean up the server -- this way there's nothing stopping us from e.g. running tests in parallel, or dealing with clearing the state on the server

Reply:
yield is nice but the issue I was pointing to was multiple servers: This now starts N http servers for N tests I believe. I don't object to that but I'll mention that it was conscious decision to not do that originally:
* I've seen plenty of issues with that in other test suites that start a server per test, especially when parallelized (think of e.g. http server not being available right away because of a hundred http servers starting at the same time)
* the startup and shutdown costs are very real once there are more tests

I'm happy to merge this though, we can always come back to this later